src/sources/ukt.rs

```diff
@@ -0,0 +1,335 @@
+//! UKT source for us to sync with Famedly Zitadel.
+
+use std::collections::HashMap;
+
+use anyhow::{bail, Context, Result};
+use chrono::Utc;
+use reqwest::Client;
+use serde::Deserialize;
+use url::Url;
+
+use crate::{config::FeatureFlags, Config, FeatureFlag};
+
+/// UKT Source
+pub struct SourceUkt {
+	/// UKT Source configuration
+	source_ukt_config: SourceUktConfig,
+	/// Optional set of features
+	feature_flags: FeatureFlags,
+	/// Reqwest client
+	client: Client,
+}
+
+impl SourceUkt {
+	/// Create a new SourceUkt instance
+	pub fn new(config: &Config) -> Result<Self> {
+		let client = Client::new();
+		Ok(Self {
+			source_ukt_config: match config.source_ukt.clone() {
+				Some(source_ukt) => source_ukt,
+				None => bail!("Endpoint configuration is missing"),
```

Comment:
> That string needs to be updated.

Done.

> Better is probably using `.unwrap_or` and a `?`.

`.unwrap_or` with `?` won't work there. I can use `.ok_or_else` as in:
```
ukt_config: config
  .sources_config
  .ukt_config
  .clone()
  .ok_or_else(|| anyhow::anyhow!("UKT configuration is missing"))?,
```
I personally prefer the match syntax here. But not going to force it. Do you want me to change all three `new` methods to use the `.ok_or_else` syntax?

Reply:
I think I primarily prefer it because it limits nesting. Deeply nested blocks like this can be hard to process. We can also pull the definition out of the struct definition and just have a variable called `ukt_config` in scope, that would likely looks nicest. But as long as the error message is correct I don't mind too much.

Whatever we do here should be done with the other sources too, yeah.

Reply:
Done.